zk_prover/src/circuits/utils.rs

```diff
@@ -27,35 +33,48 @@ use snark_verifier::{
     verifier::plonk::PlonkSuccinctVerifier,
 };
 use snark_verifier_sdk::{evm::gen_evm_proof_shplonk, CircuitExt};
-use std::{
-    fs::File,
-    io::{BufRead, BufReader, Write},
-    path::PathBuf,
-};
 
-/// Generate setup parameters for a circuit of size `k` where 2^k is the number of rows in the circuit.
-/// First checks if the trusted setup parameters are already generated and saved in the `ptau` folder with the name `hermez-raw-k`, if so, it loads them.
-/// If the trusted setup are not found in the `ptau` folder, an unsafe trusted setup is generated and saved there.
-pub fn generate_setup_params(k: u32) -> ParamsKZG<Bn256> {
-    let ptau_path = format!("ptau/hermez-raw-{}", k);
-
-    let metadata = std::fs::metadata(ptau_path.clone());
-
-    if metadata.is_err() {
-        println!("ptau file not found, generating a trusted setup of our own. If needed, download the ptau from https://github.com/han0110/halo2-kzg-srs");
-        // start timer
-        let timer = start_timer!(|| "Creating params");
-        let params = ParamsKZG::<Bn256>::setup(k, OsRng);
-        end_timer!(timer);
-        params
-    } else {
-        println!("ptau file found");
-        let timer = start_timer!(|| "Creating params");
-        let mut params_fs = File::open(ptau_path).expect("couldn't load params");
-        let params = ParamsKZG::<Bn256>::read(&mut params_fs).expect("Failed to read params");
-        end_timer!(timer);
-        params
+/// Generate setup artifacts for a circuit of size `k`, where 2^k represents the number of rows in the circuit.
+///
+/// If the trusted setup parameters are not found, the function performs an unsafe trusted setup to generate the necessary parameters
+/// If the provided `k` value is larger than the `k` value of the loaded parameters, an error is returned, as the provided `k` is too large.
+/// Otherwise, if the `k` value is smaller than the `k` value of the loaded parameters, the parameters are downsized to fit the requested `k`.
```

Comment:
I see that here you claim that you are downsizing the params but I don't see it being implmented in the function

Reply:
Ah It missed out, Fixed at [9ab6c78](https://github.com/summa-dev/summa-solvency/pull/117/commits/9ab6c786f1e4961dbfc7922b366257653bf236bd)